code/modules/antagonists/vampire/vampire_powers/dantalion_powers.dm

```diff
@@ -0,0 +1,198 @@
+		to_chat(target, "<span class='warning'>You feel the tendrils of evil invade your mind.</span>")
+		if(do_mob(user, target, 5 SECONDS))
+			if(can_enthrall(user, target))
+				handle_enthrall(user, target)
+				var/datum/spell_handler/vampire/V = custom_handler
+				var/blood_cost = V.calculate_blood_cost(vampire)
+				vampire.bloodusable -= blood_cost //we take the blood after enthralling, not before
+		else
+			revert_cast(user)
+			to_chat(user, "<span class='warning'>You or your target moved.</span>")
+
+/obj/effect/proc_holder/spell/vampire/enthrall/proc/can_enthrall(mob/living/user, mob/living/carbon/C)
+	if(!C)
+		log_runtime(EXCEPTION("target was null while trying to vampire enthrall, attacker is [user] [user.key] \ref[user]"), user)
+		return FALSE
+	if(!ishuman(C))
+		to_chat(user, "<span class='warning'>You can only enthrall sentient humanoids!</span>")
+		return FALSE
+	if(!C.mind)
+		to_chat(user, "<span class='warning'>[C.name]'s mind is not there for you to enthrall.</span>")
+		return FALSE
+
+	var/enthrall_safe = FALSE
+	for(var/obj/item/implant/mindshield/L in C)
+		if(L.implanted)
+			enthrall_safe = TRUE
+			break
+	var/datum/antagonist/vampire/V = user.mind.has_antag_datum(/datum/antagonist/vampire)
+	if(V.subclass.thrall_cap <= length(user.mind.som.serv))
+		to_chat(user, "<span class='warning'>You don't have enough power to enthrall any more people!</span>")
+		return FALSE
+	if(enthrall_safe || C.mind.has_antag_datum(/datum/antagonist/vampire) || C.mind.has_antag_datum(/datum/antagonist/mindslave))
+		C.visible_message("<span class='warning'>[C] seems to resist the takeover!</span>", "<span class='notice'>You feel a familiar sensation in your skull that quickly dissipates.</span>")
+		return FALSE
+	if(C.mind.isholy)
+		C.visible_message("<span class='warning'>[C] seems to resist the takeover!</span>", "<span class='notice'>Your faith in [SSticker.Bible_deity_name] has kept your mind clear of all evil.</span>")
+		return FALSE
+	return TRUE
+
+/obj/effect/proc_holder/spell/vampire/enthrall/proc/handle_enthrall(mob/living/user, mob/living/carbon/human/H)
+	if(!istype(H))
+		return FALSE
+
+	var/greet_text = "<b>You have been Enthralled by [user.real_name]. Follow [user.p_their()] every command.</b>"
+	H.mind.add_antag_datum(new /datum/antagonist/mindslave/thrall(user.mind, greet_text))
+	H.Stun(2)
+	user.create_log(CONVERSION_LOG, "vampire enthralled [H.real_name]")
+	H.create_log(CONVERSION_LOG, "was vampire entrhalled by [user.real_name]")
+
+/obj/effect/proc_holder/spell/vampire/thrall_commune
+	name = "Commune"
+	desc = "Talk to your thralls telepathically."
+	gain_desc = "You have gained the ability to commune with your thralls."
+
+/obj/effect/proc_holder/spell/vampire/thrall_commune/create_new_handler() //so thralls can use it
+	RETURN_TYPE(/datum/spell_handler)
```

Comment:
Is this one needed?

Reply:
It needs to not have a spell handler so that thralls can use the spell. This is what `/obj/effect/proc_holder/spell` did so I assumed that was needed

Reply:
Ah aight. This is a macro used by one of our extensions which basically says that this proc will return this type.
It only needs to be done at the base definition

Reply:
So this can just early return?

Reply:
Yep just `return` should be enough here